### ISSUE TITLE
[ts-command-line] Strange output after standing up basic app.

### ISSUE THREAD

**Issue by Jason-Terry** [microsoft/web-build-tools#1287]
Greetings, 

```BASH
jason.a.terry@CPX-TGVY1906UUE MINGW64 ~/OneDrive - Accenture/Documents/Dev/Snips/node-snips/blog-db-api
$ note -h
Method: NoteAction.onDefineParameters not implemented
usage: note [-h] [-v] <command> ...
note: error: argument "<command>": Invalid choice: C:\Program Files\nodejs\node.exe (choose from [note])
```

The above is the output I am getting after outlining a basic app.

```TypeScript
const chalk = require('chalk');
const clear = require('clear');
const figlet = require('figlet');
// const path = require('path');

import { CommandLineParser, CommandLineFlagParameter } from '@microsoft/ts-command-line'

import { NoteAction } from './actions'

export class NoteCLI extends CommandLineParser {
    
    private _verbose: CommandLineFlagParameter | undefined;

    public constructor() {
         super({
            toolFilename: 'note',
            toolDescription: 'Take notes without tabbing out!'
         });

         this.addAction(new NoteAction());
    }

    protected onDefineParameters(): void { // abstract
        this._verbose = this.defineFlagParameter({
          parameterLongName: '--verbose',
          parameterShortName: '-v',
          description: 'Show extra logging detail'
        });
      }
    
      protected onExecute(): Promise<void> { // override
        console.log("TRC: Calling Execute on Parser.")
        this.banner();
        return super.onExecute();
      }

    private banner() {
        clear();
        console.log( chalk.red( 
            figlet.textSync('note-cli', { horizontalLayout: 'full' })
        ));
    }
}
```

The above is my parser subclass, outlined in a similar way to that of the only given example.

```TypeScript
import { CommandLineAction } from "@microsoft/ts-command-line";

export class NoteAction extends CommandLineAction {

    public constructor() {
        super({
            actionName: 'note',
            summary: 'Creates a note in the default notebook',
            documentation: 'Don\'t now what more you want from me...'
          });
    }

    protected onDefineParameters(): void {
        console.error("Method: NoteAction.onDefineParameters not implemented");
    }    
    
    protected onExecute(): Promise<void> {
        console.error("Method: NoteAction.onDefineParameters not implemented");
        return new Promise((resolve, reject) => {
            // the resolve / reject functions control the fate of the promise
            // create note with given action args
            // save note to default notebook
            console.log("Writing Note!")
        });
    }


}
```

This is the only action defined.

I am really struggling with making heads or tails of this library, I am sorry if this is the wrong place but I am not finding a ton of material on how to make this work for me.

Thank you, to anyone who takes the time to read this and assist. I appreciate any help.

**Comment by octogonz** [comment]
Hey @Jason-Terry,

I tried building your code snippet above. It was missing the startup statement, so I added this:

```ts
const cli= new NoteCLI();
cli.execute();
```

My output looked like this:

```
$ node lib/start.js -h
Method: NoteAction.onDefineParameters not implemented
usage: note [-h] [-v] <command> ...

Take notes without tabbing out!

Positional arguments:
  <command>
    note         Creates a note in the default notebook

Optional arguments:
  -h, --help     Show this help message and exit.
  -v, --verbose  Show extra logging detail

For detailed help about a specific command, use: note <command> -h
```

The above seems correct to me. I assume the problem you were referring to was this line (which did not happen for me):
```
note: error: argument "<command>": Invalid choice: C:\Program Files\nodejs\node.exe (choose from [note])
```

Could you provide a more complete repro?  For example a GitHub branch and some repro steps?

**Comment by tao-cumplido** [comment]
I have a similar problem, I get the following output
```
usage: build [-h] <command> ...
build: error: argument "<command>": Invalid choice: build (choose from [])
```
I am trying to start the tool with ts-node instead of node, maybe that's the problem.
I can try to setup a minimal repo in the next days.

**Comment by tao-cumplido** [comment]
@octogonz I think I found the problem. The example in the readme passes `process.argv` directly:

```ts
const commandLine: WidgetCommandLine = new WidgetCommandLine();
commandLine.execute(process.argv);
```
I didn't notice that you run `execute()` without argument in your example. If I do it like that or pass `process.argv.slice(2)` it works as expected.

**Comment by octogonz** [comment]
Yikes...  what an insidious typo!  We need to fix those docs.  😇

**Comment by Jason-Terry** [comment]
> @octogonz I think I found the problem. The example in the readme passes `process.argv` directly:
> 
> ```ts
> const commandLine: WidgetCommandLine = new WidgetCommandLine();
> commandLine.execute(process.argv);
> ```
> 
> I didn't notice that you run `execute()` without argument in your example. If I do it like that or pass `process.argv.slice(2)` it works as expected.

Thank you so much, if this is a documentation issue I can attempt to correct it. Glad I was able to stumble on something that was actually helpful.

**Comment by octogonz** [comment]
Fixed in #1332
